webapp/src/components/confirm_modal.tsx

```diff
@@ -0,0 +1,202 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {PureComponent} from 'react';
+import {Modal} from 'react-bootstrap';
+import {FormattedMessage} from 'react-intl';
+
+export type Props = {
+
+    /*
+     * Set to show modal
+     */
+    show: boolean;
+
+    /*
+     * Title to use for the modal
+     */
+    title: React.ReactNode;
+
+    /*
+     * Message to display in the body of the modal
+     */
+    message: React.ReactNode;
+
+    /*
+     * The CSS class to apply to the confirm button
+     */
+    confirmButtonClass: string;
+
+    /*
+     * The CSS class to apply to the modal
+     */
+    modalClass?: string;
+
+    /*
+     * Text/jsx element on the confirm button
+     */
+    confirmButtonText: React.ReactNode;
+
+    /*
+     * Text/jsx element on the cancel button
+     */
+    cancelButtonText: React.ReactNode;
+
+    /*
+     * Set to show checkbox
+     */
+    showCheckbox?: boolean;
+
+    /*
+     * Text/jsx element to display with the checkbox
+     */
+    checkboxText?: string;
+
+    /*
+     * Function called when the confirm button or ENTER is pressed. Passes `true` if the checkbox is checked
+     */
+    onConfirm?: (arg0: boolean) => void;
+
+    /*
+     * Function called when the cancel button is pressed or the modal is hidden. Passes `true` if the checkbox is checked
+     */
+    onCancel?: (arg0: boolean) => void;
+
+    /**
+     * Function called when modal is dismissed
+     */
+    onExited?: () => void;
+
+    /*
+     * Set to hide the cancel button
+     */
+    hideCancel: boolean;
+
+}
+
+export type State = {
+}
+
+export default class ConfirmModal extends PureComponent<Props, State> {
+    componentDidMount() {
+        if (this.props.show) {
+            document.addEventListener('keydown', this.handleKeypress);
+        }
+    }
+
+    componentWillUnmount() {
+        document.removeEventListener('keydown', this.handleKeypress);
+    }
+
+    shouldComponentUpdate(nextProps) {
+        return nextProps.show !== this.props.show;
+    }
+
+    UNSAFE_componentWillReceiveProps(nextProps) { // eslint-disable-line camelcase
+        if (this.props.show && !nextProps.show) {
+            document.removeEventListener('keydown', this.handleKeypress);
+        } else if (!this.props.show && nextProps.show) {
+            document.addEventListener('keydown', this.handleKeypress);
+        }
+    }
```

Comment:
We should try not to introduce these `UNSAFE_*` methods if possible. 2/5 use `componentDidUpdate` instead. It takes `prevProps`, while the current (or "next") props are in `this.props`.
```
    componentDidUpdate(prevProps) {
        if (prevProps.show && !this.props.show) {
            document.removeEventListener('keydown', this.handleKeypress);
        } else if (!prevProps.show && this.props.show) {
            document.addEventListener('keydown', this.handleKeypress);
        }
    }
```